blueprints/test-framework-detector.js

```diff
@@ -12,7 +14,12 @@ module.exports = function(blueprint) {
     if ('ember-cli-qunit' in dependencies) {
       type = 'qunit';
     } else if ('ember-cli-mocha' in dependencies) {
-      type = 'mocha';
+      if (fs.existsSync(this.path + '/mocha-0.12-files') &&
+        semver.satisfies('0.12.0', dependencies['ember-cli-mocha'])) {
+        type = 'mocha-0.12';
+      } else {
+        type = 'mocha';
```

Comment:
regarding the folder names: I wanted to keep the diff small, so I decided to leave the old blueprints at their original location